pkg/dockerregistry/server/tagservice_test.go

```diff
@@ -257,24 +209,21 @@ func TestTagCreationWithoutImageStream(t *testing.T) {
 	namespace := "user"
 	repo := "app"
 	tag := "latest"
-	client := &testclient.Fake{}
-
-	serverURL, _ := url.Parse("docker.io/centos")
 
-	testImage := createTestImageReactor(t, client, serverURL, namespace, repo)
-	createTestImageStreamReactor(t, client, testImage, namespace, repo+"-another", tag)
+	_, client := registrytest.NewFakeOpenShiftWithClient()
 
 	r := newTestRepository(t, namespace, repo, testRepositoryOptions{
-		client:            client,
-		enablePullThrough: true,
+		client: client,
 	})
+
 	ts := &tagService{
 		TagService: newTestTagService(nil),
 		repo:       r,
 	}
 
-	ctx := context.Background()
-	err := ts.Tag(ctx, tag, distribution.Descriptor{Digest: digest.Digest(testImage.Name)})
+	err := ts.Tag(context.Background(), tag, distribution.Descriptor{
+		Digest: unknownBlobDigest,
```

Comment:
This test verifies that the tagging of the image fails if it's not contained in the image stream even if the image *exists*.